### PR TITLE
Training script

### PR DESCRIPTION
**Pull request by b-remy**
This pull request is made to review the code, however not test has been done yet.
- The training script can now handle any dataset, giving the specifications to the FLAGS.
- A new theoretical power spectrum has been added to match the cosmology and redshift distribution of our maps (kappatng).

### CODE REVIEW

**Comment by EiffL** [comment]
- [ ] : Remove notebooks we don't want to merge (Map and Sampling)
- [ ] : Fix the Gaussian prior in sampling/training scripts
- [ ] : @EiffL do a review

**Comment by b-remy** [comment]
I made the changes we talked about this morning. I think the PR is ready to be reviewed :)

File: jax_lensing/models/convdae.py [hunk 1]
```diff
@@ -365,7 +365,8 @@ def __init__(self,
                      channels_per_group=(32, 64, 128, 128),
                      use_projection=(True, True, True, True),
                      # 320 -> 160 -> 80 -> 40
-                     strides=(2, 2, 2, 2),
+                     # 360 -> 180 -> 90 -> 45
+                     strides=(2, 2, 2, 1),
```
Comment (EiffL) [inline jax_lensing/models/convdae.py:369]: maybe we shouldn't modify this line if for now  we are using the convdae2 for the 360x360 case

File: jax_lensing/spectral.py [hunk 2]
```diff
@@ -17,15 +17,22 @@ def radial_profile(data):
...
+  :param pixel_size: pixel_size (rad/pixel)
+  :return: ell
   :return: power spectrum
+  
   """
   data_ft = jnp.fft.fftshift(jnp.fft.fft2(map_data)) / map_data.shape[0]
   nyquist = np.int(map_data.shape[0]/2)
-  return radial_profile(jnp.real(data_ft*jnp.conj(data_ft)))[:nyquist]
+  power_spectrum_1d =  radial_profile(jnp.real(data_ft*jnp.conj(data_ft)))[:nyquist] * (pixel_size)**2
+  k = np.arange(power_spectrum_1d.shape[0])
```
Comment (EiffL) [inline jax_lensing/spectral.py:32]: hummmmm maybe we should use fftfreq for this? altough it should return the same result...

File: scripts/sample_maps.py [hunk 3]
```diff
@@ -42,27 +43,31 @@
 flags.DEFINE_string("gaussian_path", "data/massivenu/mnu0.0_Maps10_PS_theory.npy", "Path to Massive Nu power spectrum.")
 flags.DEFINE_boolean("gaussian_only", False, "Only use Gaussian score if yes.")
 flags.DEFINE_boolean("reduced_shear", False, "Apply reduced shear correction if yes.")
+flags.DEFINE_integer("map_size", 360, "Map size")
```
Comment (EiffL) [inline scripts/sample_maps.py:46]: Do we need a map_size parameter? We can deduct it from the shear map or the mask file I guess

**Review by EiffL — CHANGES_REQUESTED** [review]
Here are a few comments:

- We should have a link or script to explain how we generated the ktng_PS_theory.npy otherwise, 6 months from now, we wil have no idea where this comes from and if it is even correct ^^'

And I added a few comments, I can take care of these points myself

File: jax_lensing/spectral.py (hunk 2 above)
Comment (EiffL) [reply jax_lensing/spectral.py:32]: ok maybe not ^^'

**Comment by b-remy** [comment]
Here is the code I used to process the ` ktng_PS_theory.npy` file : https://github.com/CosmoStat/Tutorials/blob/power-spectrum/WL_power_spectrum.ipynb

File: scripts/sample_maps.py (hunk 3 above)
Comment (b-remy) [reply scripts/sample_maps.py:46]: Yes it can be deduced from the shear map or mask shape... I'll edit the file

Comment (b-remy) [reply scripts/sample_maps.py:46]: done :)

**Review by EiffL — APPROVED** [review]
Awesome thanks!
